design-documents/asynchronous-import/modularity/data-exchanging.md

```diff
@@ -0,0 +1,211 @@
+# Data exchanging with Magento instance
+
+## Modules
+
+- `AsynchronousImportDataExchangingApi`
+- `AsynchronousImportDataExchanging`
+
+## Implementation
+
+### API
+
+```php
+namespace Magento\AsynchronousImportDataExchangingApi\Api;
+
+use Magento\AsynchronousImportDataExchangingApi\Api\Data\ImportInterface;
+use Magento\Framework\Validation\ValidationException;
+
+/**
+ * Operation for exchanging import data with destination instance. Uses differect strategies for data import
+ *
+ * @api
+ */
+interface ExchangeImportDataInterface
+{
+    /**
+     * Operation for exchanging import data with destination instance
+     *
+     * @param ImportInterface $import
+     * @param array $importData
+     * @return void
+     * @throws ValidationException
+     * @throws ImportDataExchangeException
+     */
+    public function execute(ImportInterface $import, array $importData): void;
+}
+```
+
+```php
+namespace Magento\AsynchronousImportDataExchangingApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Describes how to import data
+ *
+ * @api
+ */
+interface ImportInterface extends ExtensibleDataInterface
+{
+    /**
+     * Get import uuid
+     *
+     * @return string|null
+     */
+    public function getUuid(): ?string;
+
+    /**
+     * Get import type
+     *
+     * @return string
+     */
+    public function getImportType(): string;
+
+    /**
+     * Get import behaviour
+     *
+     * @return string
+     */
+    public function getImportBehaviour(): string;
```

Comment:
What is this behavior? and why it is string... trying to understand, if there could be more better way to represent it rather than String.

Reply:
Behaviour here is an "add", "update", "delete", "replace" ....

Reply:
So on import process we can add new objects, update existed, delete some or replace.
We will start with "add" and "delete" with MVP